### PR TITLE
Add retroshare 0.6.5

### PR DESCRIPTION
**Pull request by linsui**

### CODE REVIEW

File: bucket/retroshare.json [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    ],
+    "persist": "Data",
+    "checkver": {
+        "url": "https://retroshare.cc/downloads.html",
+        "regex": "RetroShare-(?<version>[\\d.]+)-Windows-Portable-(?<build>[\\da-zA-Z\\-.]+).7z\">Retroshare \\(Portable\\)"
+    },
+    "autoupdate": {
+        "url": "https://github.com/RetroShare/RetroShare/releases/download/v$version/RetroShare-$version-Windows-Portable-$matchBuild.7z",
+        "hash": {
+            "url": "https://github.com/RetroShare/RetroShare/releases/download/v$version/chksums.txt.asc"
```
Comment (Ash258) [inline bucket/retroshare.json]:
```suggestion
            "url": "$baseurl/chksums.txt.asc"
```

File: bucket/retroshare.json [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    "shortcuts": [
+        [
+            "retroshare.exe",
+            "RetroShare"
+        ]
+    ],
+    "persist": "Data",
+    "checkver": {
+        "url": "https://retroshare.cc/downloads.html",
+        "regex": "RetroShare-(?<version>[\\d.]+)-Windows-Portable-(?<build>[\\da-zA-Z\\-.]+).7z\">Retroshare \\(Portable\\)"
```
Comment (Ash258) [inline bucket/retroshare.json]:
```suggestion
        "regex": "RetroShare-([\\d.]+)-Windows-Portable-(?<build>[\\w-.]+).7z\">Retroshare \\(Portable\\)"
```

Comment (linsui) [reply bucket/retroshare.json]: I find there are tons manifest use `?<version>` :joy:

Comment (Ash258) [reply bucket/retroshare.json]: It is irrelevant when it is firts captured group.

Comment (linsui) [reply bucket/retroshare.json]: Yeah, so those manifests make me confused. There is even such manifest. :joy:  https://github.com/lukesampson/scoop-extras/blob/a84b257fd9636d02295b48c3fd32826487ca9bd3/bucket/graphicsgale.json#L21

Comment (Ash258) [reply bucket/retroshare.json]: This could be for historical reason.

**Pull request merged**